### PR TITLE
Frontend Sandbox: Create a plugin sandbox enable registry. Use enable list instead of disable list

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Modifies the Frontend Sandbox so it uses an enable list instead of a disable list.

Additionally moves the logic to enable the sandbox for a plugin to a check function that can be overwritten. This will allow grafana extensions to determine if a plugin should run inside the sandbox or not

**Why do we need this feature?**

To make it possible to run specific plugins inside the sandbox

**Who is this feature for?**

Grafana users that want to run plugins inside the frontend sandbox

**Special notes for your reviewer:**

notice this PR is renaming an existing configuration option but it is not considered a breaking change since this feature flag was never used outside of our own testing environment.

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -404,7 +404,7 @@ angular_support_enabled = false
 csrf_always_check = false
 
 # Comma-separated list of plugins ids that won't be loaded inside the frontend sandbox
```
Comment (andresmgot) [inline conf/defaults.ini:406]: You need to update the comment. Also, I'd mention this is behind a feature flag

File: public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export async function shouldLoadPluginInFrontendSandbox({
+  isAngular,
+  pluginId,
+}: SandboxEligibilityCheckParams): Promise<boolean> {
+  // basic check if the plugin is eligible for the sandbox
+  if (!(await isPluginFrontendSandboxElegible({ isAngular, pluginId }))) {
+    return false;
+  }
```
Comment (andresmgot) [inline public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts:30]: nit: why don't you set this as a `checker`?

File: public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  return true;
+}
+
+/**
+ * Check if the plugin is enabled for the sandbox via configuration.
+ */
+async function isPluginFrontendSandboxEnabled({ pluginId }: SandboxEligibilityCheckParams): Promise<boolean> {
+  return Boolean(config.enableFrontendSandboxForPlugins?.includes(pluginId));
+}
```
Comment (andresmgot) [inline public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts:83]: also nit: not sure why this is in a different checker than isPluginFrontendSandboxElegible

File: public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ * We allow core extensions to register their own
+ * sandbox enabled checks.
+ */
+const sandboxEnabledChecks: SandboxEnabledCheck[] = [isPluginFrontendSandboxEnabled];
+
+export function addSandboxEnabledCheck(checker: SandboxEnabledCheck) {
+  sandboxEnabledChecks.push(checker);
+}
+
+export async function shouldLoadPluginInFrontendSandbox({
```
Comment (andresmgot) [inline public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts:23]: are there tests for this?

**Review by andresmgot — COMMENTED** [review]
Looking good to me. Can you also open a PR with an extension to check this works end to end? Doesn't need to do the real thing, just return `true`, to verify that the init and registration stuff works as expected.

File: public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts (hunk 2 above)
Comment (academo) [reply public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts:30]: I thought about it and decided it to separate the concept of "eligibility" with the concept of "enabled". a plugin could be enabled by not be elegible to run in the sandbox. e.g: the user puts an angular plugin in the list of enabled plugins, this won't run.

File: public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts (hunk 3 above)
Comment (academo) [reply public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts:83]: just for the sake of keeping the concept of elegible and enabled apart. Eventually we'll grow this function to have more checks to determine if a plugin is enabled or not, so I decided to keep them separated from the start.

File: public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts (hunk 4 above)
Comment (academo) [reply public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts:23]: there are e2e tests for this. I can add some unit tests

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Comment by academo** [comment]
> Looking good to me. Can you also open a PR with an extension to check this works end to end? Doesn't need to do the real thing, just return `true`, to verify that the init and registration stuff works as expected.

I'll be opening a PR with an extension soon. It is a in a separate repository and I need this merged first.

**Review by xnyo — APPROVED** [review]
LGTM, awesome work! :rocket:

File: conf/sample.ini [hunk 5]
```diff
@@ -409,7 +409,7 @@
 ;csrf_always_check = false
 
 # Comma-separated list of plugins ids that won't be loaded inside the frontend sandbox
```
Comment (andresmgot) [inline conf/sample.ini:411]: this comment is still not updated

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 6]
```diff
@@ -729,10 +729,9 @@ List of allowed headers to be set by the user. Suggested to use for if authentic
 
 Set to `true` to execute the CSRF check even if the login cookie is not in a request (default `false`).
 
-### disable_frontend_sandbox_for_plugins
+### enable_frontend_sandbox_for_plugins
 
-Comma-separated list of plugins ids that won't be loaded inside the frontend sandbox. It is recommended to only use this
-option for plugins that are known to have problems running inside the frontend sandbox.
+Comma-separated list of plugins ids that will be loaded inside the frontend sandbox.
```
Comment (andresmgot) [inline docs/sources/setup-grafana/configure-grafana/_index.md:734]: maybe add here a comment about the feature flag?

**Review by andresmgot — APPROVED** [review]
LGTM just a couple of comments missing

File: e2e/old-arch/panels-suite/frontend-sandbox-panel.spec.ts
Comment (academo) [inline e2e/old-arch/panels-suite/frontend-sandbox-panel.spec.ts:1]: These tests are duplicated from the ones that were skipped outside. I removed these and removed the skip from the duplicates.

**Pull request merged**